### DISCUSSION TITLE
Support for OpenSource Models

### DISCUSSION THREAD

**Discussion by workflowanalytics** [namankhurpia/Easy-open-ai#24] [Ideas]
Hi, thanks for providing a open source approach to using the OpenAi models.  
Some of the newer open models exceed Chat performance over chatgpt-3.5 (Mixtral in particular, https://arxiv.org/abs/2401.04088), 
but otherwise may not have support for other OpenAi models.  Many of these models utilize openai's chat api for compatibility to OpenAi, so calling them vrs OpenAi appears to be dooable.
Wondering if the expanding the keys.txt file to a CSV list of attributes necessary to support calls to compatible Chat api's.
To support a dynamic call capability, a reference of some form to the associated row would need to be passed to the calling methods.  Cheezy approach, but looking for a way to avoid a database.  Would be happy to assist but would want to do so in the context of senior developer's recommended approach.

**Comment by namankhurpia** [comment]
So just to jot it down - you wanna make sure we have the capability of using multiple keys in case of multiple OPENAI API calls, I believe I already created something like this - its called concurrency, you can check this out - 

https://github.com/namankhurpia/Easy-open-ai?tab=readme-ov-file#EMMC-EasyOpenAI-Multithreaded-MultiKey-ChatCompletionAPI

I think we can develop a lot on top of it, let me know if you wanna work together, I have some good ideas @workflowanalytics

**Reply by workflowanalytics** [reply]
Yes, the concurrent calls to different models is part of the solution.  
But the provider is only to openai and to the openai's URL to submit the request to.
The keys.txt file only provides the authentication key, and the method calls (chat, image, etc) specify the model.

In addition, I would like to supply a reference to an authentication key specific to the provider and the URL of the provider
the call should be made to. As many providers support the openai api, my thought was that your system could utilize non-openai provider models that support the openai curl format. Possibly this could be passed in this call or, as I was thinking earlier, as one of two values in the keys.txt file (providerUrl, authenticationKey).  The model I am most interested in is the Mixtral 7B model.

Thanks for your feedback on this
ChatCompletionResponse response = new EasyopenaiService(new DAOImpl()).chatCompletion("OPENAI_KEY",request);
